src/compiler/checker.ts

```diff
@@ -7352,7 +7619,8 @@ namespace ts {
             containsNonWideningType?: boolean;
             containsString?: boolean;
             containsNumber?: boolean;
-            containsStringOrNumberLiteral?: boolean;
+            containsESSymbol?: boolean;
+            containsStringOrNumberLiteralOrUniqueESSymbol?: boolean;

```

Comment:
We also consider `true` and `false` to be `Literal`, even though we don't handle them the same way as string and numeric literals.

Reply:
Here we're just concerned with knowing whether we need an extra pass to eliminate literal types that are subsumed by some non-union base type. We don't need to do anything special for `true` and `false` because their base type is already a union type.